comms/src/parse/Requests.ts

```diff
@@ -0,0 +1,123 @@
+import { Instruction } from "../debug/Instructions";
+
+export interface StackValue {
+    idx: number,
+    type: string;
+    value: number | bigint;
+}
+
+export interface CallbackMapping {
+    callbackid: string;
+    tableIndexes: number[]
+}
+
+export interface InterruptEvent {
+    topic: string;
+    payload: string;
+}
+
+export interface Frame {
+    type: number;
+    fidx: string;
+    sp: number;
+    fp: number;
+    block_key: number;
+    ra: number;
+    idx: number;
+}
+
+export interface Table {
+    max: number;
```

Comment:
We should indeed respect the wasm spec but the interfaces and fields that I introduce here need to have this structure to make out-of-place possible. The interfaces are to enable the reconstruction of a pulled state into a local VM.

Now maybe to avoid confusion between Wasm specific state and WARDuino specific state. we can rename those interfaces to have a prefix e.g., WARDuinoTable

Reply:
No I wouldn't rename to WARDuinoTable. I think it is fine to have additional fields if they have a good motivation.
I would just like the fields to have better names, because it is not clear what they mean right now.

I guess max is the maximum size of the table, but what is init? That sounds like an internal WARDuino thing that is the same for all tables, so might not even be needed in the snapshot for OOP debugging.

This isn't high priority right now. I don't think it stands in the way of merging this PR.

Reply:
I understand about the need for better naming but if we do change that then we can no longer easily parse the printed VM state as this:

```TS
const state: State = JSON.parser(str);

```

This because currently the interface field names match the printed json property fields. If we want better names for the fields, it is possible but then we need to add code to do create objects with the desired fields.

I think `init` is the initial table size. I'm not sure whether we need that value when interpreting some Wasm but let's keep it for now. As I'm not directly sure of the implications of removing it. At first sight, we do not need it for OOP.

Reply:
ok whatever keep it. But it is not like the interfaces type check anything.